### PR TITLE
Add docs for non-shaded JDBC driver JAR

### PR DESCRIPTION
**Pull request by mosabua**
## Description

See title and changes.

## Additional context and related issues

Follow up to https://github.com/trinodb/trino/pull/22098

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/client/jdbc.md [hunk 1]
```diff
@@ -21,7 +21,10 @@ Versions before 350 are not supported.
 
 Download {maven_download}`jdbc` and add it to the classpath of your Java application.
 
-The driver is also available from Maven Central:
+The driver is available from Maven Central as a full JAR archive that contains
```
Comment (wendigo) [inline docs/src/main/sphinx/client/jdbc.md:24]: s/full/shaded ?

Comment (mosabua) [reply docs/src/main/sphinx/client/jdbc.md:24]: Lots of people wont know what "shaded" means

Comment (wendigo) [reply docs/src/main/sphinx/client/jdbc.md:24]: fat jar then?

Comment (mosabua) [reply docs/src/main/sphinx/client/jdbc.md:24]: Sure .. kinda weird term but it is more commonly used.

Comment (mosabua) [reply docs/src/main/sphinx/client/jdbc.md]: Updated now.

**Review by gaeljw — APPROVED** [review]
LGTM.

I was wondering if we should give some concrete use cases of one or the other but I think it's not that easy to give a "definitive" information. Some might use the fat JAR in a context where someone else might prefer the non shaded one.

**Comment by wendigo** [comment]
@gaeljw there are basically two use cases:

- JDBC driver used in some BI/IDE tool which usually don't have a maven resolver hence fat jar is used because it is standalone,

- JDBC driver used in the application (as a Maven/Gradle/SBT dependency), in this case the user can choose between fat-jar and non-shaded artifact, the latter usually requires settling on the version bounds/resolving conflicts which could be a tricky since the Trino depends on the latest versions of the libraries we depend upon

**Review by wendigo — APPROVED** [review]

**Pull request merged**